src/cargo/util/config.rs

```diff
@@ -31,11 +31,7 @@ pub struct Config {
 }
 
 impl Config {
-    pub fn new(shell: MultiShell) -> CargoResult<Config> {
-        let cwd = try!(env::current_dir().chain_error(|| {
```

Comment:
If this function is already fallible, how come this moved out to the caller? I think this is only called from one place, right?